File: 프로그래머스/lv1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
def solution(n, m):
    minNum = min(n, m)
    maxNum = max(n, m)

    def gcd(a, b):
        while(a != 0):
            n = b % a
            b = a
            a = n
        return b

    gcd = gcd(minNum,maxNum)
    lcm = (minNum*maxNum) // gcd
    return [gcd,lcm]

File: 프로그래머스/lv1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.js
function solution(n, m) {
    const num1 = Math.max(n, m);
    const num2 = Math.min(n, m);
    const gcd = (a, b) => (b > 0 ? gcd(b, a % b) : a);
    const gcd_result = gcd(num1, num2);
    
    return [gcd_result, parseInt(n*m/gcd_result)]
}